docs/testplan/IPv4-Port-Based-DHCP-Server-test-plan.md

```diff
@@ -0,0 +1,196 @@
+  * Use CLI to modify `DHCP_SERVER_IPV4_PORT` / `DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS` table to see whether `dhcp_server:/etc/kea/kea-dhcp4.conf` change as expected.
+
+#### test_dhcp_server_config_vlan_change
+
+* **Test objective**
+
+   To test vlan configuration change scenario.
+
+* **Test detail**
+
+  * Modify vlan ip in `VLAN_INTERFACE` table, change to another subnet. Check `dhcp_server:/etc/kea/kea-dhcp4.conf`, expect no interface configured.
+  * Modify vlan ip back and remove vlan member in `VLAN_MEMBER` table that was configured in `DHCP_SERVER_IPV4_PORT` table, expect that configuration doesn't contain removed port.
+
+#### test_dhcp_server_critical_process
+
+* **Test objective**
+
+   To test critical processes crush scenario.
+
+* **Test detail**
+
+   * Kill processes in `dhcp_relay:/etc/supervisor/critical_processes` and `dhcp_server:/etc/supervisor/critical_processes` to see whether dhcp_server and dhcp_relay container restart.
+   * Can refer to `tests/process_monitoring/test_critical_process_monitoring.py`
+
+### Test Module #2 test_dhcp_server_multi_vlan.py
+
+#### Common setup
+
+* Enable dhcp_server feature.
+* Apply DHCP Server related configuration for multiple vlans (Suggest use GCU).
+* Install isc-dhcp-client in ptf.
+
+#### Common teardown
+
+* Config reload, remove dhcp_server container.
+* Uninstall isc-dhcp-client in ptf.
+
+#### test_dhcp_server_multi_vlan
+
+* **Test objective**
+
+   To test ip assign in multiple vlan scenario.
+
+* **Test detail**
+
+  * Use GCU to apply different VLAN configuration (VLAN / VLAN_MEMBER / VLAN_INTERFACE).
+  * Check whether dhcpmon / dhcrelay processes running status is consistent with configuration.
+  * Check whether `dhcp_server:/etc/kea/kea-dhcp4.conf` is consistent with configuration.
+  * Use dhclient in ptf to check whether configured port can get ip and non-configured port cannot get ip.
+
+### Test Module #3 test_dhcp_server_stress.py
+
+#### Common setup
+
+* Enable dhcp_server feature, and then add DHCP Server configuration.
+
+#### Common teardown
+
+* Config reload, remove dhcp_server container.
+
+#### test_dhcp_server_stress
+
+* **Test objective**
+
+   To test ip assign with flooding packets.
+
+* **Test detail**
+
+   * Send flooding (100/s) DHCP discover packets in ptf and verify offer packets receive (whether receive and receive time) in ptf side.
+   * Send flooding (100/s) DHCP request packets in ptf and verify ack packets receive (whether receive and receive time) in ptf side.
+   * During test, increasement of CPU and memory utilization don't exceed threshold.
+
+### Test Module #4 test_dhcp_server_smart_switch.py
+
+#### Common setup
+
+* Enable dhcp_server feature.
+* Set `DEVICE_METADATA|localhost|subtype` to "SmartSwitch".
+* Create a bridge in DUT and add 4 interface into it.
```

Comment:
Simulate a smart switch?
I suppose this test need to run on real smart switch unless we have a virtual smart switch